ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/replyboxview/ReplyBoxFragmentTest.java

```diff
@@ -0,0 +1,350 @@
+
+    @Mock
+    private KeyEvent keyEvent;
+
+    @Mock
+    private Intent intent;
+
+    @Mock
+    private AppCompatActivity appCompatActivity;
+
+    @Mock
+    private Editable editable;
+
+    @Mock
+    private ImageButton imageButton;
+
+    @Mock
+    private ReplyBoxContract.ReplyBoxListener listener;
+
+    @Captor
+    private ArgumentCaptor<String> stringArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<Integer> integerArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<Boolean> booleanArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<ReplyBoxContract.ReplyBoxListener> replyBoxListenerArgumentCaptor;
+
+    @Rule
+    public final ErrorCollector errorCollector = new ErrorCollector();
+
+    @Test
+    public void onCreate() {
+        //Act
+        replyBoxFragment.onCreate(bundle);
+        final ReplyBoxContract.Presenter mPresenter =
+                Whitebox.getInternalState(replyBoxFragment, "mPresenter");
+
+        //Assert
+        assertNotNull(mPresenter);
+    }
+
+    @Test
+    public void onCreateView() {
+        //Arrange
+        when(layoutInflater.inflate(R.layout.ko__fragment_reply_box, viewGroup, false)).thenReturn(view);
+
+
+        //Act
+        replyBoxFragment.onCreateView(layoutInflater, viewGroup, bundle);
+        final View mRoot = Whitebox.getInternalState(replyBoxFragment, "mRoot");
+
+        //Assert
+        assertEquals(view, mRoot);
+    }
+
+    @Test
+    public void onViewCreated() throws Exception {
+        //Arrange
+        final String dummyPowerMessage = "dummyPowerMessage";
+        suppress(methods(Fragment.class, "startActivity"));
+        mockStatic(KayakoCreditsHelper.class);
+        mockStatic(Uri.class);
+        uri = mock(Uri.class);
+        Whitebox.setInternalState(replyBoxFragment, "mRoot", view);
+        Whitebox.setInternalState(replyBoxFragment, "mPresenter", presenter);
+        when(view.findViewById(R.id.ko__reply_box_send_button)).thenReturn(view);
+        when(view.findViewById(R.id.reply_box_edittext)).thenReturn(editText);
+        when(keyEvent.getKeyCode()).thenReturn(KeyEvent.KEYCODE_ENTER);
+        when(view.findViewById(R.id.ko__reply_box_attach_button)).thenReturn(view);
+        when(view.findViewById(R.id.ko__reply_box_credits_by)).thenReturn(textView);
+        when(KayakoCreditsHelper.getPoweredByMessage()).thenReturn(dummyPowerMessage);
+        when(view.findViewById(R.id.ko__reply_box_credits_kayako)).thenReturn(textView);
+        when(KayakoCreditsHelper.getLink()).thenReturn("link");
+        when(Uri.parse("link")).thenReturn(uri);
+        whenNew(Intent.class).withArguments(Intent.ACTION_VIEW, uri).thenReturn(intent);
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+                Object arguments[] = invocation.getArguments();
+                View.OnClickListener listener = (View.OnClickListener)arguments[0];
```

Comment:
❗️ This code is using the class field `listener` as a local variable. Local variables should not shadow class fields. Shadowing fields with a local variable or with a method parameter is a bad practice that reduces code readability, as it is confusing to know whether the field or the variable is being used. Consider renaming this local variable.  
 

<a id="6b19eb00-53ca-11e8-b777-83ccc384eaae" href="none" codereview-uuid-reserved-tag />

Reply:
⚠️  Formatting is inconsistent. Applying standard Java formatting (or project specific style) to all files before committing maintains code style consistency and support the five "easy": easy to read, easy to understand, easy to modify, easy to maintain, easy to enhance. Please use the IDE format capabilities before committing.  

**[More details](https://confluence.devfactory.com/x/U48-Ew)**
          
## The solution in this case

```JAVA
View.OnClickListener listenerLocal = (View.OnClickListener) arguments[0];
                             ^^^^^                         ^ 
```

To have a consistent and uniform formatting, consider using the IDE formatting capabilities before committing the code.
 

<a id="73ce01a0-53ca-11e8-b777-83ccc384eaae" href="none" codereview-uuid-reserved-tag />